server/views/applications/pages/risk-of-serious-harm/risk-management-arrangements.njk

```diff
@@ -86,4 +86,40 @@
       ) 
   }}
 
+{% endblock %}
+
+{% block extraScripts %}
+  <script type="text/javascript" nonce="{{ cspNonce }}">
+    function setCheckedToFalseByIds(ids) {
+      ids.forEach(id => {     
+        document.getElementById(id).checked = false;
+      })
+    }
+
+    function clearValueByIds(ids) {
+      ids.forEach(id => {
+        document.getElementById(id).value = ''
+      })
+    }
+
+    document.addEventListener('DOMContentLoaded', function () {
+      // 'no' checkbox sets the others to unchecked
+      document
+        .getElementById('arrangements-5')
+        .addEventListener('click', () => {
+          setCheckedToFalseByIds(['arrangements', 'arrangements-2', 'arrangements-3'])
+          clearValueByIds(['mappaDetails', 'maracDetails', 'iomDetails'])
+        });
+      // clicking any arrangement checkbox sets 'no' to unchecked
+      document
+        .getElementById('arrangements')
+        .addEventListener('click', () => setCheckedToFalseByIds(['arrangements-5']));
+      document
+        .getElementById('arrangements-2')
+        .addEventListener('click', () => setCheckedToFalseByIds(['arrangements-5']));
+      document
+        .getElementById('arrangements-3')
+        .addEventListener('click', () => setCheckedToFalseByIds(['arrangements-5']));
+    });
```

Comment:
Is this a scenario where selecting by class could add clarity and reliability?

Am I right to think we could end up with something simpler like e.g. 

```
    document.addEventListener('DOMContentLoaded', function () {
      // 'no' checkbox sets the others to unchecked
      document
        .getElementsByClassName('no-arrangements')
        .addEventListener('click', () => {
          clearAnySelectedArrangements()
        });
      // clicking any arrangement checkbox sets 'no' to unchecked
      document
        .getElementsByClassName('arrangement')
        .addEventListener('click', () => resetNoArrangementsChoice());
    });
```

Apologies in advance if I'm way off the mark here!

Reply:
thanks Ed, that's a really good shout!

Reply:
unfortunately it didn't seem possible to add a `class` to an `item` according to the nunjucks options https://design-system.service.gov.uk/components/checkboxes/, but I've set a data attribute that will achieve a similar thing and allow us to select all three in one go.